src/redux/sagas/appSaga.ts

```diff
@@ -81,8 +94,23 @@ function* updateState() {
   ]);
 }
 
-export function* appWatcherSaga(): Generator<any, any, any> {
+export function* appWatcherSaga(): SagaGenerator {
   yield takeLatest(INIT_APP, initAppSaga);
+  // Restore user settings before signing in
+  yield takeLatest(INIT_APP_SUCCESS, function* () {
+    yield all([
```

Comment:
suggestion: use `fork` here to run the restore sagas? `all` will cancel all tasks if one fails, `fork` lets to run each "restore" independently

Reply:
I would keep `all` in that case, which I think is better suited. The error propagation works similarly for both `all` and attached `forks`. A Saga aborts as soon as its main body of instructions throws an error or an uncaught error is raised by one of its attached forks. https://redux-saga.js.org/docs/advanced/ForkModel#error-propagation

`fork` best use cases are cancellation feature and running a non-blocking task (but it doesn't matter here. `takeLatest` is non-blocking and there is no other code to execute after the restore sagas in `takeLatest` worker). https://stackoverflow.com/a/42961360